hydromt/models/v1/grid.py

```diff
@@ -0,0 +1,722 @@
+        gdal_compliant: bool = False,
+        rename_dims: bool = False,
+        force_sn: bool = False,
+        **kwargs,
+    ) -> None:
+        """Write model grid data to netcdf file at <root>/<fn>.
+
+        key-word arguments are passed to :py:meth:`~hydromt.models.Model.write_nc`
+
+        Parameters
+        ----------
+        fn : str, optional
+            filename relative to model root, by default 'grid/grid.nc'
+        **kwargs : dict
+            Additional keyword arguments to be passed to the `write_nc` method.
+        gdal_compliant : bool, optional
+            If True, write grid data in a way that is compatible with GDAL,
+            by default False
+        rename_dims: bool, optional
+            If True and gdal_compliant, rename x_dim and y_dim to standard names
+            depending on the CRS (x/y for projected and lat/lon for geographic).
+        force_sn: bool, optional
+            If True and gdal_compliant, forces the dataset to have
+            South -> North orientation.
+        """
+        if len(self._data) == 0:
+            self.logger.debug("No grid data found, skip writing.")
+        else:
+            self._assert_write_mode()  # Model class method
+            # write_nc requires dict - use dummy 'grid' key
+            self.write_nc(
+                {"grid": self._data},
+                fn,
+                gdal_compliant=gdal_compliant,
+                rename_dims=rename_dims,
+                force_sn=force_sn,
+                **kwargs,
+            )
+
+    def read(
+        self, fn: str = "grid/grid.nc", read: bool = True, write: bool = False, **kwargs
+    ) -> None:
+        """Read model grid data at <root>/<fn> and add to grid property.
+
+        key-word arguments are passed to :py:meth:`~hydromt.models.Model.read_nc`
+
+        Parameters
+        ----------
+        fn : str, optional
+            filename relative to model root, by default 'grid/grid.nc'
+        **kwargs : dict
+            Additional keyword arguments to be passed to the `read_nc` method.
+        """
+        self._assert_read_mode()  # TODO Model class method, could be refactored to a model_utils function
+        self._initialize_grid(skip_read=True)
+
+        # Load grid data in r+ mode to allow overwritting netcdf files
+        if read and write:
+            kwargs["load"] = True
+        loaded_nc_files = self.read_nc(
+            fn, single_var_as_array=False, **kwargs
+        )  # TODO: decide where read_nc should be placed
+        for ds in loaded_nc_files.values():
+            self.set(ds)
+
+    def create(self):
+        """Create grid."""
+        raise NotImplementedError
+
+    @property
+    def model(self):
+        """Returns model reference."""
+        # Access the Model instance through the weak reference TODO: discuss if this is needed
+        return self._model_ref()
+
+    @property
+    def data(self):
+        """Returns data."""
+        return self._data
```

Comment:
we're not doing anything special like lazy loading to these ones so I wonder if these need to be properties.